core/templates/services/playthrough.service.ts

```
@@ -308,24 +334,33 @@ export class PlaythroughService {
       this.recordedLearnerActions !== null);
   }
 
-  private hasRecordingFinished(): boolean {
-    return (
-      this.hasRecordingBegun() &&
-      this.recordedLearnerActions.length > 1 &&
-      this.recordedLearnerActions[this.recordedLearnerActions.length - 1]
-        .actionType === AppConstants.ACTION_TYPE_EXPLORATION_QUIT);
+  // Return Undefined if Learner has not begun recording.
+  private hasRecordingFinished(): boolean | undefined {
+    if (this.recordedLearnerActions !== null) {
```

Should we rather throw an error if it is `null`?

---

Done